reactor-core/src/test/java/reactor/core/publisher/OnDiscardShouldNotLeakTest.java

```diff
@@ -58,6 +61,58 @@ public class OnDiscardShouldNotLeakTest {
 	// add DiscardScenarios here to test more operators
 	private static final DiscardScenario[] SCENARIOS = new DiscardScenario[] {
 			DiscardScenario.allFluxSourceArray("merge", 4, Flux::merge),
+			DiscardScenario.allFluxSourceArray("zip", 4,
+					sources -> {
+						Publisher<Tracked>[] sources1 = sources.toArray(new Publisher[0]);
+						return Flux.zip(Tuples::fromArray, sources1)
+						           .doOnNext(l -> {
+							           for (Object o : (Iterable<Object>) l) {
+								           ((Tracked) o).release();
+							           }
+						           })
+						           .then(Mono.just(Tracked.RELEASED));
+					}),
+			DiscardScenario.allFluxSourceArray("zipFusedAll", 4,
+					sources -> {
+						Publisher<Tracked>[] sources1 = sources.stream().map(s -> s.publishOn(Schedulers.immediate())).toArray(Publisher[]::new);
+						return Flux.zip(Tuples::fromArray, sources1)
+						           .doOnNext(l -> {
+							           for (Object o : (Iterable<Object>) l) {
+								           ((Tracked) o).release();
+							           }
+						           })
+						           .then(Mono.just(Tracked.RELEASED));
+					}),
+			DiscardScenario.allFluxSourceArray("zipFusedRandom", 4,
+					sources -> {
+						Publisher<Tracked>[] sources1 = sources.stream()
+						                                       .map(s ->
+								                                       ThreadLocalRandom.current()
+								                                                        .nextBoolean() ?
+										                                       s.publishOn(
+												                                       Schedulers.immediate()) :
+										                                       s)
+						                                       .toArray(Publisher[]::new);
+						return Flux.zip(Tuples::fromArray, sources1)
+						           .doOnNext(l -> {
+							           for (Object o : (Iterable<Object>) l) {
+								           ((Tracked) o).release();
+							           }
+						           })
+						           .then(Mono.just(Tracked.RELEASED));
+					}),
+			DiscardScenario.allFluxSourceArray("zipScalar", 4,
+					sources -> {
+						Publisher<Tracked>[] sources1 =
+								Stream.concat(sources.stream(), Stream.of(Mono.just(Tracked.RELEASED))).toArray(Publisher[]::new);
```

Comment:
Scenarios are defined statically, so within a call, we can not get access to the allocator which holds all the references to `Tracked` objects. In this particular case, we initiate a specific path for `FluxZip`, which uses `ZipSingleCoordinator` instead of the default `ZipCoordinator`. Thus, it is not that important for now, although this can be improved in a separate PR so the rules API can get access to allocator upon the rule installation

Reply:
I see. Now I understand it's not vital to validate the scalar value is released. Thanks.